spec/models/manageiq/providers/microsoft/infra_manager/provision_workflow_spec.rb

```diff
@@ -16,6 +16,27 @@
     described_class.new({}, admin.userid)
   end
 
+  describe "#load_hosts_vlans" do
+    let(:host)        { FactoryGirl.create(:host_with_ref) }
+    let(:switch)      { FactoryGirl.create(:switch) }
+    let(:lan_parent)  { FactoryGirl.create(:lan) }
```

Comment:
The test actually fails sporadically if the parent lan is also on the switch which it would be IRL

Maybe change the `lan_for_host` to detect on parent not being nil?